src/library_gl.js

```diff
@@ -919,12 +921,34 @@ var LibraryGL = {
     if (GL.stringCache[name_]) return GL.stringCache[name_];
     var ret;
     switch(name_) {
+      case 0x1F03 /* GL_EXTENSIONS */:
+        var exts = GLctx.getSupportedExtensions();
+        var gl_exts = [];
+        for (var i = 0; i < exts.length; ++i) {
+          gl_exts.push(exts[i]);
+          gl_exts.push("GL_" + exts[i]);
+        }
+        ret = allocate(intArrayFromString(gl_exts.join(' ')), 'i8', ALLOC_NORMAL);
+        break;
       case 0x1F00 /* GL_VENDOR */:
       case 0x1F01 /* GL_RENDERER */:
       case 0x9245 /* UNMASKED_VENDOR_WEBGL */:
       case 0x9246 /* UNMASKED_RENDERER_WEBGL */:
-        ret = allocate(intArrayFromString(GLctx.getParameter(name_)), 'i8', ALLOC_NORMAL);
+#if !GL_EMULATE_GLES_VERSION_STRING_FORMAT
+      case 0x1F02 /* GL_VERSION */:
+      case 0x8B8C /* GL_SHADING_LANGUAGE_VERSION */:
+#endif
+        var s = GLctx.getParameter(name_);
+        if (!s) {
+          GL.recordError(0x0500/*GL_INVALID_ENUM*/);
+#if GL_ASSERTIONS
+          err('GL_INVALID_ENUM in glGetString: Received empty parameter for query name ' + name_ + '!'); // This occurs e.g. if one attempts GL_UNMASKED_VENDOR_WEBGL when it is not supported.
+#endif
+        }
+        ret = allocate(intArrayFromString(s), 'i8', ALLOC_NORMAL);
         break;
+
+#if GL_EMULATE_GLES_VERSION_STRING_FORMAT
```

Comment:
Most codebases do not care about the syntax in which WebGL version is returned, so the code to map WebGL version strings to OpenGL ES version strings just wastes space.